openedx/openedx-platform
# fix: allow content libraries to be created through Django admin

## Description

Previously, CMS Django admin would demand that at least one LTIConfig be associated with the new library. This is not necessary, but Django thought it was, because the `blank` flag was not enabled on the ContentLibrary model's `authored_lti_config` many-to-many field.

This should not affect the experimental blockstore-based library authoring frontend, which already allows libraries to be created without any authorized_lti_configs.

This will not affect the existing modulestore-based library authoring frontend, since it doesn't even use any of the models in question.

A migration is included with this commit, but it should NOT change MySQL schema, since `blank` is a Django-level validation flag.

## Testing instructions

Ensure you have an existing Blockstore Bundle created. Note its UUID.

Start up CMS. In CMS admin, go to create a Blockstore-Based Content Library: http://studio.local.overhang.io:8001/admin/content_libraries/contentlibrary/ . Make up a slug, paste in your Bundle UUID, and choose an Org (create one new if you need to). Hit "Save".

![image](https://user-images.githubusercontent.com/3628148/217094956-a7bb5079-9709-4c05-9d9e-126d139f083f.png)

Before this bugfix, you would see:

![image](https://user-images.githubusercontent.com/3628148/217095018-9c647ff7-f2fb-405e-9d30-df759ffe3718.png)

With the fix, you should see:

![image](https://user-images.githubusercontent.com/3628148/217095358-616394ea-aaac-416b-af00-16d566cf1e97.png)

## Deadline

None

## Other information

N/A
